app/src/main/java/com/example/flightticket/LoginActivity.java

```diff
@@ -17,6 +42,53 @@ public static Intent intentFactory(Context context) {
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
+
+        loginEmail = findViewById(R.id.loginEmail);
+        loginPassword = findViewById(R.id.loginPassword);
+        loginBtn = findViewById(R.id.loginBtn);
+
+
+        userDAO = Room.databaseBuilder(this, UserDatabase.class, UserDatabase.DB_NAME)
+                .allowMainThreadQueries()
+                .fallbackToDestructiveMigration()
+                .build().getUserDAO();
+
+        usernames = userDAO.getUsernames();
+
+        loginBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                emailInput = loginEmail.getText().toString();
+                passwordInput = loginPassword.getText().toString();
+                User loginUser = userDAO.getUserByUsername(emailInput);
+                if (validateUser(emailInput,usernames)&&validatePassword(passwordInput,loginUser.getPassword())){
+                    Intent i= new Intent(getApplicationContext(), MainActivity.class);
+                    startActivity(i);
```

Comment:
This should take you to the HomeActivity rather than Main. Other than that, looks good to me.

Reply:
We do not need main activity then?

Reply:
Intent now directs to HomeActivity